client/my-sites/media/details.jsx

```diff
@@ -0,0 +1,247 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import page from 'page';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import { EditorMediaModalDetail } from 'post-editor/media-modal/detail';
+import { getMimePrefix, getMimeType } from 'lib/media/utils';
+import ImageEditor from 'blocks/image-editor';
+import VideoEditor from 'blocks/video-editor';
+import MediaActions from 'lib/media/actions';
+
+import MediaLibrarySelectedStore from 'lib/media/library-selected-store';
+import accept from 'lib/accept';
+
+class MediaDetails extends Component {
```

Comment:
I think separating this in a separate component could be material for a separate PR itself. It can be done and will be a good improvement regardless of which approach you'll go with for implementing single media page - `<Media />` is already huge and could use some decoupling and separation of concerns.

Then, once the `MediaDetails` are in a separate component, we could slowly start introducing changes (like the `edit` vs `view` mode that we're introducing in this PR. This will allow us to test more granularly and assure that we're not breaking existing functionality.